src/impression.js

```diff
@@ -14,31 +14,58 @@
  * limitations under the License.
  */
 
-import {user} from './log';
+import {dev, user} from './log';
 import {isExperimentOn} from './experiments';
 import {viewerForDoc} from './viewer';
 import {xhrFor} from './xhr';
+import {
+  isProxyOrigin,
+  parseUrl,
+  parseQueryString,
+  addParamsToUrl,
+} from './url';
+import {getMode} from './mode';
 
+let trackImpressionPromise = null;
+
+/**
+ * A function to get the trackImpressionPromise;
+ * @return {!Promise}
+ */
+export function getTrackImpressionPromise() {
+  return dev().assert(trackImpressionPromise);
+}
 
 /**
  * Emit a HTTP request to a destination defined on the incoming URL.
  * Protected by experiment.
  * @param {!Window} win
  */
 export function maybeTrackImpression(win) {
+  let resolveImpression;
+
+  trackImpressionPromise = new Promise(resolve => {
```

Comment:
`ForTesting` function added. But why can't we just export `resolveImpression` for testing?